src/modules/ejector/ejector.py

```diff
@@ -177,6 +171,15 @@ def get_validators_to_eject(self, blockstamp: ReferenceBlockStamp) -> list[tuple
     def _is_paused(self, blockstamp: ReferenceBlockStamp) -> bool:
         return self.report_contract.functions.isPaused().call(block_identifier=blockstamp.block_hash)
 
+    def is_reporting_allowed(self, blockstamp: ReferenceBlockStamp) -> bool:
+        on_pause = self._is_paused(blockstamp)
+        CONTRACT_ON_PAUSE.set(on_pause)
+        if on_pause:
```

Comment:
Maybe
`logger.info({'msg': 'Fetch on pause', 'value': on_pause})`
`return not on_pause`